{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T21:37:53.801664Z",
     "iopub.status.busy": "2023-11-08T21:37:53.801475Z",
     "iopub.status.idle": "2023-11-08T21:37:53.890279Z",
     "shell.execute_reply": "2023-11-08T21:37:53.889716Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T21:37:53.892613Z",
     "iopub.status.busy": "2023-11-08T21:37:53.892453Z",
     "iopub.status.idle": "2023-11-08T23:28:14.869128Z",
     "shell.execute_reply": "2023-11-08T23:28:14.868543Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:37:54.078708: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35033\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:39.365837: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.377586: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.378381: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.381776: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.388703: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.389615: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.471044: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.471242: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.471406: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:39.471546: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 16:39:39.471892: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:40.388000: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.388243: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.388414: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.388627: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.388864: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 16:39:40.389007: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 16:39:40.389035: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 16:39:40.411026: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 16:39:40.681215: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/kernel/Assign' id:3481 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/kernel, lstm_21/lstm_cell_21/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:39:40.814862: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:39:40.848662: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31519, 95)\n",
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:45.304373: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign' id:17552 op device:{requested: '', assigned: ''} def:{{{node training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v, training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:39:49.640592: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 16:39:52.176095: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 16:39:52.189790: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 2.9548"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 16:40:13.316887: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.83476, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 29s 917us/sample - loss: 2.9548 - val_loss: 1.8348\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.7309\n",
      "Epoch 2: val_loss improved from 1.83476 to 1.56367, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.7309 - val_loss: 1.5637\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5740\n",
      "Epoch 3: val_loss improved from 1.56367 to 1.49363, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.5740 - val_loss: 1.4936\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5266\n",
      "Epoch 4: val_loss improved from 1.49363 to 1.46463, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 645us/sample - loss: 1.5266 - val_loss: 1.4646\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5043\n",
      "Epoch 5: val_loss improved from 1.46463 to 1.45009, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 665us/sample - loss: 1.5043 - val_loss: 1.4501\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4866\n",
      "Epoch 6: val_loss improved from 1.45009 to 1.44614, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 684us/sample - loss: 1.4866 - val_loss: 1.4461\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4786\n",
      "Epoch 7: val_loss improved from 1.44614 to 1.43420, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.4786 - val_loss: 1.4342\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4739\n",
      "Epoch 8: val_loss improved from 1.43420 to 1.42649, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.4739 - val_loss: 1.4265\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4668\n",
      "Epoch 9: val_loss improved from 1.42649 to 1.41721, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 704us/sample - loss: 1.4668 - val_loss: 1.4172\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4604\n",
      "Epoch 10: val_loss improved from 1.41721 to 1.41317, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 633us/sample - loss: 1.4604 - val_loss: 1.4132\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4670\n",
      "Epoch 11: val_loss did not improve from 1.41317\n",
      "31519/31519 [==============================] - 19s 609us/sample - loss: 1.4670 - val_loss: 1.4175\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4527\n",
      "Epoch 12: val_loss did not improve from 1.41317\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.4527 - val_loss: 1.4141\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4535\n",
      "Epoch 13: val_loss improved from 1.41317 to 1.41204, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 620us/sample - loss: 1.4535 - val_loss: 1.4120\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4513\n",
      "Epoch 14: val_loss did not improve from 1.41204\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.4513 - val_loss: 1.4226\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4401\n",
      "Epoch 15: val_loss improved from 1.41204 to 1.40962, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 616us/sample - loss: 1.4401 - val_loss: 1.4096\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4353\n",
      "Epoch 16: val_loss improved from 1.40962 to 1.40067, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 618us/sample - loss: 1.4353 - val_loss: 1.4007\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5166\n",
      "Epoch 17: val_loss improved from 1.40067 to 1.40058, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.5166 - val_loss: 1.4006\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4837\n",
      "Epoch 18: val_loss improved from 1.40058 to 1.39970, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 648us/sample - loss: 1.4837 - val_loss: 1.3997\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4652\n",
      "Epoch 19: val_loss did not improve from 1.39970\n",
      "31519/31519 [==============================] - 21s 655us/sample - loss: 1.4652 - val_loss: 1.4014\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4431\n",
      "Epoch 20: val_loss did not improve from 1.39970\n",
      "31519/31519 [==============================] - 22s 710us/sample - loss: 1.4431 - val_loss: 1.4008\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4726\n",
      "Epoch 21: val_loss did not improve from 1.39970\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 1.4726 - val_loss: 1.4139\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4367\n",
      "Epoch 22: val_loss did not improve from 1.39970\n",
      "31519/31519 [==============================] - 21s 658us/sample - loss: 1.4367 - val_loss: 1.4007\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4447\n",
      "Epoch 23: val_loss did not improve from 1.39970\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.4447 - val_loss: 1.4040\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4301\n",
      "Epoch 24: val_loss improved from 1.39970 to 1.39918, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.4301 - val_loss: 1.3992\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4194\n",
      "Epoch 25: val_loss improved from 1.39918 to 1.39117, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 652us/sample - loss: 1.4194 - val_loss: 1.3912\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4149\n",
      "Epoch 26: val_loss improved from 1.39117 to 1.38592, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 616us/sample - loss: 1.4149 - val_loss: 1.3859\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4157\n",
      "Epoch 27: val_loss did not improve from 1.38592\n",
      "31519/31519 [==============================] - 19s 611us/sample - loss: 1.4157 - val_loss: 1.3860\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4093\n",
      "Epoch 28: val_loss did not improve from 1.38592\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.4093 - val_loss: 1.3886\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4064\n",
      "Epoch 29: val_loss improved from 1.38592 to 1.38492, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 668us/sample - loss: 1.4064 - val_loss: 1.3849\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4054\n",
      "Epoch 30: val_loss improved from 1.38492 to 1.38202, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.4054 - val_loss: 1.3820\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4004\n",
      "Epoch 31: val_loss improved from 1.38202 to 1.38080, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.4004 - val_loss: 1.3808\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3996\n",
      "Epoch 32: val_loss improved from 1.38080 to 1.37975, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 1.3996 - val_loss: 1.3797\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3990\n",
      "Epoch 33: val_loss improved from 1.37975 to 1.37616, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 1.3990 - val_loss: 1.3762\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3967\n",
      "Epoch 34: val_loss improved from 1.37616 to 1.37389, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.3967 - val_loss: 1.3739\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3945\n",
      "Epoch 35: val_loss improved from 1.37389 to 1.37379, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 713us/sample - loss: 1.3945 - val_loss: 1.3738\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3907\n",
      "Epoch 36: val_loss did not improve from 1.37379\n",
      "31519/31519 [==============================] - 22s 707us/sample - loss: 1.3907 - val_loss: 1.3769\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3889\n",
      "Epoch 37: val_loss improved from 1.37379 to 1.37198, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 680us/sample - loss: 1.3889 - val_loss: 1.3720\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3882\n",
      "Epoch 38: val_loss improved from 1.37198 to 1.37128, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 23s 717us/sample - loss: 1.3882 - val_loss: 1.3713\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3880\n",
      "Epoch 39: val_loss improved from 1.37128 to 1.36849, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3880 - val_loss: 1.3685\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3861\n",
      "Epoch 40: val_loss did not improve from 1.36849\n",
      "31519/31519 [==============================] - 20s 633us/sample - loss: 1.3861 - val_loss: 1.3726\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3820\n",
      "Epoch 41: val_loss improved from 1.36849 to 1.36415, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 711us/sample - loss: 1.3820 - val_loss: 1.3641\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3829\n",
      "Epoch 42: val_loss improved from 1.36415 to 1.36216, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 711us/sample - loss: 1.3829 - val_loss: 1.3622\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3809\n",
      "Epoch 43: val_loss did not improve from 1.36216\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3809 - val_loss: 1.3686\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 44: val_loss did not improve from 1.36216\n",
      "31519/31519 [==============================] - 19s 599us/sample - loss: 1.3798 - val_loss: 1.3634\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3775\n",
      "Epoch 45: val_loss did not improve from 1.36216\n",
      "31519/31519 [==============================] - 19s 604us/sample - loss: 1.3775 - val_loss: 1.3627\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3779\n",
      "Epoch 46: val_loss did not improve from 1.36216\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3779 - val_loss: 1.3656\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3780\n",
      "Epoch 47: val_loss improved from 1.36216 to 1.35938, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.3780 - val_loss: 1.3594\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3739\n",
      "Epoch 48: val_loss did not improve from 1.35938\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 1.3739 - val_loss: 1.3686\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3726\n",
      "Epoch 49: val_loss improved from 1.35938 to 1.35774, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 655us/sample - loss: 1.3726 - val_loss: 1.3577\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3729\n",
      "Epoch 50: val_loss improved from 1.35774 to 1.35601, saving model to ./checkpoints/unknown_person_baseline_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.3729 - val_loss: 1.3560\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:57:06.044339: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_25_1/lstm_cell_62/bias/Assign' id:22671 op device:{requested: '', assigned: ''} def:{{{node lstm_25_1/lstm_cell_62/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_25_1/lstm_cell_62/bias, lstm_25_1/lstm_cell_62/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:57:07.765288: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_4/bias/v/Assign' id:25858 op device:{requested: '', assigned: ''} def:{{{node conv2d_4/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_4/bias/v, conv2d_4/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 16:57:09.118626: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1788, 1968)\n",
      "(1812, 1968)\n",
      "(1788, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 7.0834789739713555, 2: 6.129131437532148, 4: 8.6392466298936, 5: 5.399634317971829, 6: 4.699411941286318, 8: 8.552974474495478, 9: 6.79716274226205, 10: 8.366878417949378, 11: 7.21943811188274, 12: 9.61195073451728, 13: 6.916538225421666, 17: 9.393091898191518, 19: 8.159486760939215, 21: 10.0, 22: 2.4102030009959594, 25: 8.880614655855052, 27: 5.241161862570428, 28: 7.192737086179737, 29: 1.0}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2792784/1157781576.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:01:11.449616: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 9.9021\n",
      "Epoch 1: val_loss improved from inf to 1.37092, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 26s 821us/sample - loss: 9.9021 - val_loss: 1.3709\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8330\n",
      "Epoch 2: val_loss improved from 1.37092 to 1.36507, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 9.8330 - val_loss: 1.3651\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7995\n",
      "Epoch 3: val_loss did not improve from 1.36507\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 9.7995 - val_loss: 1.3822\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7941\n",
      "Epoch 4: val_loss did not improve from 1.36507\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 9.7941 - val_loss: 1.3663\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7504\n",
      "Epoch 5: val_loss improved from 1.36507 to 1.35933, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 9.7504 - val_loss: 1.3593\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7258\n",
      "Epoch 6: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 9.7258 - val_loss: 1.3596\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7130\n",
      "Epoch 7: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 22s 707us/sample - loss: 9.7130 - val_loss: 1.3696\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7080\n",
      "Epoch 8: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 9.7080 - val_loss: 1.3609\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6698\n",
      "Epoch 9: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 9.6698 - val_loss: 1.3633\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6513\n",
      "Epoch 10: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 20s 649us/sample - loss: 9.6513 - val_loss: 1.3629\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6541\n",
      "Epoch 11: val_loss did not improve from 1.35933\n",
      "31519/31519 [==============================] - 21s 651us/sample - loss: 9.6541 - val_loss: 1.3684\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6430\n",
      "Epoch 12: val_loss improved from 1.35933 to 1.35795, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 657us/sample - loss: 9.6430 - val_loss: 1.3580\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6237\n",
      "Epoch 13: val_loss improved from 1.35795 to 1.35575, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 632us/sample - loss: 9.6237 - val_loss: 1.3557\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6213\n",
      "Epoch 14: val_loss improved from 1.35575 to 1.35010, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 9.6213 - val_loss: 1.3501\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6094\n",
      "Epoch 15: val_loss did not improve from 1.35010\n",
      "31519/31519 [==============================] - 20s 637us/sample - loss: 9.6094 - val_loss: 1.3535\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5949\n",
      "Epoch 16: val_loss did not improve from 1.35010\n",
      "31519/31519 [==============================] - 20s 619us/sample - loss: 9.5949 - val_loss: 1.3534\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5792\n",
      "Epoch 17: val_loss improved from 1.35010 to 1.34856, saving model to ./checkpoints/unknown_person_p26_20.h5\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 9.5792 - val_loss: 1.3486\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5589\n",
      "Epoch 18: val_loss did not improve from 1.34856\n",
      "31519/31519 [==============================] - 19s 598us/sample - loss: 9.5589 - val_loss: 1.3556\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5511\n",
      "Epoch 19: val_loss did not improve from 1.34856\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 9.5511 - val_loss: 1.3548\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5731\n",
      "Epoch 20: val_loss did not improve from 1.34856\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 9.5731 - val_loss: 1.3493\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:08:08.383639: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_2/lstm_cell_86/bias/Assign' id:39987 op device:{requested: '', assigned: ''} def:{{{node lstm_12_2/lstm_cell_86/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_2/lstm_cell_86/bias, lstm_12_2/lstm_cell_86/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:08:10.810606: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_22_2/lstm_cell_96/recurrent_kernel/v/Assign' id:45628 op device:{requested: '', assigned: ''} def:{{{node lstm_22_2/lstm_cell_96/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_22_2/lstm_cell_96/recurrent_kernel/v, lstm_22_2/lstm_cell_96/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:08:15.903956: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:08:28.401470: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3705"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:08:51.390350: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.35693, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 29s 931us/sample - loss: 1.3705 - val_loss: 1.3569\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3672\n",
      "Epoch 2: val_loss did not improve from 1.35693\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.3672 - val_loss: 1.3581\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3663\n",
      "Epoch 3: val_loss improved from 1.35693 to 1.35213, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.3663 - val_loss: 1.3521\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3662\n",
      "Epoch 4: val_loss did not improve from 1.35213\n",
      "31519/31519 [==============================] - 19s 603us/sample - loss: 1.3662 - val_loss: 1.3579\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3653\n",
      "Epoch 5: val_loss did not improve from 1.35213\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.3653 - val_loss: 1.3566\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3627\n",
      "Epoch 6: val_loss did not improve from 1.35213\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3627 - val_loss: 1.3543\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3621\n",
      "Epoch 7: val_loss did not improve from 1.35213\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3621 - val_loss: 1.3532\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3623\n",
      "Epoch 8: val_loss improved from 1.35213 to 1.35118, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3623 - val_loss: 1.3512\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3602\n",
      "Epoch 9: val_loss improved from 1.35118 to 1.34578, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3602 - val_loss: 1.3458\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3607\n",
      "Epoch 10: val_loss did not improve from 1.34578\n",
      "31519/31519 [==============================] - 21s 657us/sample - loss: 1.3607 - val_loss: 1.3484\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3574\n",
      "Epoch 11: val_loss did not improve from 1.34578\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3574 - val_loss: 1.3477\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3608\n",
      "Epoch 12: val_loss did not improve from 1.34578\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.3608 - val_loss: 1.3529\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3535\n",
      "Epoch 13: val_loss did not improve from 1.34578\n",
      "31519/31519 [==============================] - 22s 684us/sample - loss: 1.3535 - val_loss: 1.3513\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3537\n",
      "Epoch 14: val_loss did not improve from 1.34578\n",
      "31519/31519 [==============================] - 21s 651us/sample - loss: 1.3537 - val_loss: 1.3478\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 15: val_loss improved from 1.34578 to 1.34410, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 20s 634us/sample - loss: 1.3519 - val_loss: 1.3441\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3515\n",
      "Epoch 16: val_loss did not improve from 1.34410\n",
      "31519/31519 [==============================] - 18s 587us/sample - loss: 1.3515 - val_loss: 1.3464\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3513\n",
      "Epoch 17: val_loss improved from 1.34410 to 1.34242, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 656us/sample - loss: 1.3513 - val_loss: 1.3424\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 18: val_loss did not improve from 1.34242\n",
      "31519/31519 [==============================] - 22s 688us/sample - loss: 1.3498 - val_loss: 1.3466\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3501\n",
      "Epoch 19: val_loss improved from 1.34242 to 1.34205, saving model to ./checkpoints/unknown_person_baseline_2_p26_20.h5\n",
      "31519/31519 [==============================] - 21s 653us/sample - loss: 1.3501 - val_loss: 1.3420\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3478\n",
      "Epoch 20: val_loss did not improve from 1.34205\n",
      "31519/31519 [==============================] - 19s 604us/sample - loss: 1.3478 - val_loss: 1.3480\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3486\n",
      "Epoch 1: val_loss improved from inf to 1.34357, saving model to ./checkpoints/unknown_person_baseline_p26_21.h5\n",
      "31519/31519 [==============================] - 19s 607us/sample - loss: 1.3486 - val_loss: 1.3436\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3438\n",
      "Epoch 2: val_loss improved from 1.34357 to 1.34274, saving model to ./checkpoints/unknown_person_baseline_p26_21.h5\n",
      "31519/31519 [==============================] - 19s 609us/sample - loss: 1.3438 - val_loss: 1.3427\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3434\n",
      "Epoch 3: val_loss did not improve from 1.34274\n",
      "31519/31519 [==============================] - 20s 624us/sample - loss: 1.3434 - val_loss: 1.3455\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3424\n",
      "Epoch 4: val_loss improved from 1.34274 to 1.33703, saving model to ./checkpoints/unknown_person_baseline_p26_21.h5\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.3424 - val_loss: 1.3370\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3392\n",
      "Epoch 5: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3392 - val_loss: 1.3388\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3429\n",
      "Epoch 6: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3429 - val_loss: 1.3495\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3404\n",
      "Epoch 7: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 676us/sample - loss: 1.3404 - val_loss: 1.3394\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 8: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 599us/sample - loss: 1.3508 - val_loss: 1.3423\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3555\n",
      "Epoch 9: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 20s 633us/sample - loss: 1.3555 - val_loss: 1.3440\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5013\n",
      "Epoch 10: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.5013 - val_loss: 1.3644\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3653\n",
      "Epoch 11: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3653 - val_loss: 1.3547\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3646\n",
      "Epoch 12: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 658us/sample - loss: 1.3646 - val_loss: 1.3509\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3612\n",
      "Epoch 13: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 684us/sample - loss: 1.3612 - val_loss: 1.3587\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3599\n",
      "Epoch 14: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3599 - val_loss: 1.3480\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3702\n",
      "Epoch 15: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.3702 - val_loss: 1.3664\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5157\n",
      "Epoch 16: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.5157 - val_loss: 1.3830\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3934\n",
      "Epoch 17: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 669us/sample - loss: 1.3934 - val_loss: 1.3580\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3859\n",
      "Epoch 18: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 653us/sample - loss: 1.3859 - val_loss: 1.3533\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3896\n",
      "Epoch 19: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 670us/sample - loss: 1.3896 - val_loss: 1.3605\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3895\n",
      "Epoch 20: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3895 - val_loss: 1.3631\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3927\n",
      "Epoch 21: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3927 - val_loss: 1.3675\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3905\n",
      "Epoch 22: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 662us/sample - loss: 1.3905 - val_loss: 1.3672\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3878\n",
      "Epoch 23: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 1.3878 - val_loss: 1.3630\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3848\n",
      "Epoch 24: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3848 - val_loss: 1.3627\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3780\n",
      "Epoch 25: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 20s 637us/sample - loss: 1.3780 - val_loss: 1.3724\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3691\n",
      "Epoch 26: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 1.3691 - val_loss: 1.3631\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3705\n",
      "Epoch 27: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 696us/sample - loss: 1.3705 - val_loss: 1.3513\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3683\n",
      "Epoch 28: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 669us/sample - loss: 1.3683 - val_loss: 1.3567\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3653\n",
      "Epoch 29: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 20s 641us/sample - loss: 1.3653 - val_loss: 1.3667\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3638\n",
      "Epoch 30: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 672us/sample - loss: 1.3638 - val_loss: 1.3563\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3581\n",
      "Epoch 31: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3581 - val_loss: 1.3620\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3614\n",
      "Epoch 32: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 614us/sample - loss: 1.3614 - val_loss: 1.3538\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3580\n",
      "Epoch 33: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 21s 669us/sample - loss: 1.3580 - val_loss: 1.3518\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3559\n",
      "Epoch 34: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 593us/sample - loss: 1.3559 - val_loss: 1.3516\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3560\n",
      "Epoch 35: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 597us/sample - loss: 1.3560 - val_loss: 1.3506\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3535\n",
      "Epoch 36: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 590us/sample - loss: 1.3535 - val_loss: 1.3496\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3556\n",
      "Epoch 37: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 588us/sample - loss: 1.3556 - val_loss: 1.3551\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3504\n",
      "Epoch 38: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 596us/sample - loss: 1.3504 - val_loss: 1.3460\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 39: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 591us/sample - loss: 1.3519 - val_loss: 1.3474\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3490\n",
      "Epoch 40: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 600us/sample - loss: 1.3490 - val_loss: 1.3534\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 41: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.3489 - val_loss: 1.3465\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3484\n",
      "Epoch 42: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3484 - val_loss: 1.3465\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3449\n",
      "Epoch 43: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 596us/sample - loss: 1.3449 - val_loss: 1.3397\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3447\n",
      "Epoch 44: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 610us/sample - loss: 1.3447 - val_loss: 1.3486\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3443\n",
      "Epoch 45: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 697us/sample - loss: 1.3443 - val_loss: 1.3424\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3422\n",
      "Epoch 46: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3422 - val_loss: 1.3473\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3420\n",
      "Epoch 47: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3420 - val_loss: 1.3430\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3438\n",
      "Epoch 48: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 1.3438 - val_loss: 1.3525\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3424\n",
      "Epoch 49: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 594us/sample - loss: 1.3424 - val_loss: 1.3492\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3412\n",
      "Epoch 50: val_loss did not improve from 1.33703\n",
      "31519/31519 [==============================] - 19s 590us/sample - loss: 1.3412 - val_loss: 1.3453\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:32:23.786592: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_22_3/lstm_cell_133/kernel/Assign' id:60920 op device:{requested: '', assigned: ''} def:{{{node lstm_22_3/lstm_cell_133/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_22_3/lstm_cell_133/kernel, lstm_22_3/lstm_cell_133/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:32:27.067626: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_26_3/lstm_cell_137/kernel/v/Assign' id:65043 op device:{requested: '', assigned: ''} def:{{{node lstm_26_3/lstm_cell_137/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_26_3/lstm_cell_137/kernel/v, lstm_26_3/lstm_cell_137/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:32:30.164586: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1788, 1968)\n",
      "(1812, 1968)\n",
      "(1788, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 7.169701996536802, 2: 6.062723489588454, 4: 8.761159586892319, 5: 5.583795972929689, 6: 4.8155020644651705, 8: 8.5937091132963, 9: 6.796121556871362, 10: 8.42042998244985, 11: 7.081702346043758, 12: 9.706241138334615, 13: 6.741821575181451, 17: 9.480307006736357, 19: 8.270055698876774, 21: 10.0, 22: 2.6790035539350265, 25: 8.86727744380326, 27: 5.154502780282301, 28: 7.099617612939042, 29: 1.0}\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:36:20.612258: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 9.8001\n",
      "Epoch 1: val_loss improved from inf to 1.36019, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 29s 904us/sample - loss: 9.8001 - val_loss: 1.3602\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7911\n",
      "Epoch 2: val_loss improved from 1.36019 to 1.35889, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 22s 705us/sample - loss: 9.7911 - val_loss: 1.3589\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.7136\n",
      "Epoch 3: val_loss improved from 1.35889 to 1.35858, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 20s 649us/sample - loss: 9.7136 - val_loss: 1.3586\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6738\n",
      "Epoch 4: val_loss improved from 1.35858 to 1.34831, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 21s 680us/sample - loss: 9.6738 - val_loss: 1.3483\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6863\n",
      "Epoch 5: val_loss did not improve from 1.34831\n",
      "31519/31519 [==============================] - 22s 710us/sample - loss: 9.6863 - val_loss: 1.3502\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6544\n",
      "Epoch 6: val_loss did not improve from 1.34831\n",
      "31519/31519 [==============================] - 20s 647us/sample - loss: 9.6544 - val_loss: 1.3564\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6141\n",
      "Epoch 7: val_loss did not improve from 1.34831\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 9.6141 - val_loss: 1.3538\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5806\n",
      "Epoch 8: val_loss did not improve from 1.34831\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 9.5806 - val_loss: 1.3513\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5801\n",
      "Epoch 9: val_loss did not improve from 1.34831\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 9.5801 - val_loss: 1.3496\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5813\n",
      "Epoch 10: val_loss improved from 1.34831 to 1.34600, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 21s 678us/sample - loss: 9.5813 - val_loss: 1.3460\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5919\n",
      "Epoch 11: val_loss did not improve from 1.34600\n",
      "31519/31519 [==============================] - 22s 713us/sample - loss: 9.5919 - val_loss: 1.3463\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5661\n",
      "Epoch 12: val_loss did not improve from 1.34600\n",
      "31519/31519 [==============================] - 20s 646us/sample - loss: 9.5661 - val_loss: 1.3479\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5511\n",
      "Epoch 13: val_loss did not improve from 1.34600\n",
      "31519/31519 [==============================] - 19s 608us/sample - loss: 9.5511 - val_loss: 1.3468\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5317\n",
      "Epoch 14: val_loss did not improve from 1.34600\n",
      "31519/31519 [==============================] - 19s 605us/sample - loss: 9.5317 - val_loss: 1.3502\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5412\n",
      "Epoch 15: val_loss did not improve from 1.34600\n",
      "31519/31519 [==============================] - 19s 606us/sample - loss: 9.5412 - val_loss: 1.3552\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5312\n",
      "Epoch 16: val_loss improved from 1.34600 to 1.34254, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 9.5312 - val_loss: 1.3425\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5179\n",
      "Epoch 17: val_loss did not improve from 1.34254\n",
      "31519/31519 [==============================] - 19s 604us/sample - loss: 9.5179 - val_loss: 1.3451\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5098\n",
      "Epoch 18: val_loss did not improve from 1.34254\n",
      "31519/31519 [==============================] - 19s 603us/sample - loss: 9.5098 - val_loss: 1.3533\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4995\n",
      "Epoch 19: val_loss did not improve from 1.34254\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 9.4995 - val_loss: 1.3435\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4955\n",
      "Epoch 20: val_loss improved from 1.34254 to 1.34238, saving model to ./checkpoints/unknown_person_p26_21.h5\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 9.4955 - val_loss: 1.3424\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:18.143428: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_19_4/lstm_cell_167/bias/Assign' id:79865 op device:{requested: '', assigned: ''} def:{{{node lstm_19_4/lstm_cell_167/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_19_4/lstm_cell_167/bias, lstm_19_4/lstm_cell_167/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 17:43:22.285558: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_2_4/kernel/m/Assign' id:83989 op device:{requested: '', assigned: ''} def:{{{node dense_2_4/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_2_4/kernel/m, dense_2_4/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:28.886990: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:43:50.534899: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3449"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 17:44:13.314802: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34275, saving model to ./checkpoints/unknown_person_baseline_2_p26_21.h5\n",
      "31519/31519 [==============================] - 32s 1ms/sample - loss: 1.3449 - val_loss: 1.3427\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3419\n",
      "Epoch 2: val_loss did not improve from 1.34275\n",
      "31519/31519 [==============================] - 19s 612us/sample - loss: 1.3419 - val_loss: 1.3436\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3406\n",
      "Epoch 3: val_loss did not improve from 1.34275\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.3406 - val_loss: 1.3448\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3430\n",
      "Epoch 4: val_loss did not improve from 1.34275\n",
      "31519/31519 [==============================] - 19s 606us/sample - loss: 1.3430 - val_loss: 1.3471\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3371\n",
      "Epoch 5: val_loss improved from 1.34275 to 1.33834, saving model to ./checkpoints/unknown_person_baseline_2_p26_21.h5\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3371 - val_loss: 1.3383\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3390\n",
      "Epoch 6: val_loss did not improve from 1.33834\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3390 - val_loss: 1.3496\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3372\n",
      "Epoch 7: val_loss did not improve from 1.33834\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3372 - val_loss: 1.3473\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3369\n",
      "Epoch 8: val_loss did not improve from 1.33834\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.3369 - val_loss: 1.3456\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3366\n",
      "Epoch 9: val_loss did not improve from 1.33834\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.3366 - val_loss: 1.3407\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3357\n",
      "Epoch 10: val_loss improved from 1.33834 to 1.33786, saving model to ./checkpoints/unknown_person_baseline_2_p26_21.h5\n",
      "31519/31519 [==============================] - 22s 713us/sample - loss: 1.3357 - val_loss: 1.3379\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3341\n",
      "Epoch 11: val_loss did not improve from 1.33786\n",
      "31519/31519 [==============================] - 20s 632us/sample - loss: 1.3341 - val_loss: 1.3410\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3329\n",
      "Epoch 12: val_loss did not improve from 1.33786\n",
      "31519/31519 [==============================] - 21s 658us/sample - loss: 1.3329 - val_loss: 1.3412\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3317\n",
      "Epoch 13: val_loss improved from 1.33786 to 1.33774, saving model to ./checkpoints/unknown_person_baseline_2_p26_21.h5\n",
      "31519/31519 [==============================] - 23s 714us/sample - loss: 1.3317 - val_loss: 1.3377\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3329\n",
      "Epoch 14: val_loss did not improve from 1.33774\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 1.3329 - val_loss: 1.3449\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3287\n",
      "Epoch 15: val_loss improved from 1.33774 to 1.33648, saving model to ./checkpoints/unknown_person_baseline_2_p26_21.h5\n",
      "31519/31519 [==============================] - 20s 625us/sample - loss: 1.3287 - val_loss: 1.3365\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3288\n",
      "Epoch 16: val_loss did not improve from 1.33648\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3288 - val_loss: 1.3401\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3287\n",
      "Epoch 17: val_loss did not improve from 1.33648\n",
      "31519/31519 [==============================] - 20s 647us/sample - loss: 1.3287 - val_loss: 1.3381\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3270\n",
      "Epoch 18: val_loss did not improve from 1.33648\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.3270 - val_loss: 1.3423\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3283\n",
      "Epoch 19: val_loss did not improve from 1.33648\n",
      "31519/31519 [==============================] - 22s 691us/sample - loss: 1.3283 - val_loss: 1.3399\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3250\n",
      "Epoch 20: val_loss did not improve from 1.33648\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3250 - val_loss: 1.3380\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3247\n",
      "Epoch 1: val_loss improved from inf to 1.33809, saving model to ./checkpoints/unknown_person_baseline_p26_22.h5\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 1.3247 - val_loss: 1.3381\n",
      "Epoch 2/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3255\n",
      "Epoch 2: val_loss improved from 1.33809 to 1.33483, saving model to ./checkpoints/unknown_person_baseline_p26_22.h5\n",
      "31519/31519 [==============================] - 20s 626us/sample - loss: 1.3255 - val_loss: 1.3348\n",
      "Epoch 3/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3252\n",
      "Epoch 3: val_loss did not improve from 1.33483\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 1.3252 - val_loss: 1.3412\n",
      "Epoch 4/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3227\n",
      "Epoch 4: val_loss did not improve from 1.33483\n",
      "31519/31519 [==============================] - 22s 693us/sample - loss: 1.3227 - val_loss: 1.3368\n",
      "Epoch 5/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3205\n",
      "Epoch 5: val_loss improved from 1.33483 to 1.33426, saving model to ./checkpoints/unknown_person_baseline_p26_22.h5\n",
      "31519/31519 [==============================] - 23s 715us/sample - loss: 1.3205 - val_loss: 1.3343\n",
      "Epoch 6/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3217\n",
      "Epoch 6: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3217 - val_loss: 1.3381\n",
      "Epoch 7/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3219\n",
      "Epoch 7: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 1.3219 - val_loss: 1.3368\n",
      "Epoch 8/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3320\n",
      "Epoch 8: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 1.3320 - val_loss: 1.3361\n",
      "Epoch 9/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3412\n",
      "Epoch 9: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3412 - val_loss: 1.3359\n",
      "Epoch 10/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3554\n",
      "Epoch 10: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 679us/sample - loss: 1.3554 - val_loss: 1.3349\n",
      "Epoch 11/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3594\n",
      "Epoch 11: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 702us/sample - loss: 1.3594 - val_loss: 1.3460\n",
      "Epoch 12/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3428\n",
      "Epoch 12: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 678us/sample - loss: 1.3428 - val_loss: 1.3499\n",
      "Epoch 13/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3532\n",
      "Epoch 13: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 668us/sample - loss: 1.3532 - val_loss: 1.3416\n",
      "Epoch 14/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3521\n",
      "Epoch 14: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 699us/sample - loss: 1.3521 - val_loss: 1.3450\n",
      "Epoch 15/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4093\n",
      "Epoch 15: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 630us/sample - loss: 1.4093 - val_loss: 1.3533\n",
      "Epoch 16/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3713\n",
      "Epoch 16: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 676us/sample - loss: 1.3713 - val_loss: 1.3572\n",
      "Epoch 17/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3745\n",
      "Epoch 17: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 19s 609us/sample - loss: 1.3745 - val_loss: 1.3556\n",
      "Epoch 18/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3769\n",
      "Epoch 18: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 659us/sample - loss: 1.3769 - val_loss: 1.3542\n",
      "Epoch 19/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3783\n",
      "Epoch 19: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 680us/sample - loss: 1.3783 - val_loss: 1.3575\n",
      "Epoch 20/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4186\n",
      "Epoch 20: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 685us/sample - loss: 1.4186 - val_loss: 1.3540\n",
      "Epoch 21/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.5896\n",
      "Epoch 21: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 662us/sample - loss: 1.5896 - val_loss: 1.3800\n",
      "Epoch 22/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4761\n",
      "Epoch 22: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.4761 - val_loss: 1.3763\n",
      "Epoch 23/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4208\n",
      "Epoch 23: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.4208 - val_loss: 1.3709\n",
      "Epoch 24/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.4019\n",
      "Epoch 24: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.4019 - val_loss: 1.3644\n",
      "Epoch 25/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3920\n",
      "Epoch 25: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.3920 - val_loss: 1.3641\n",
      "Epoch 26/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3865\n",
      "Epoch 26: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 710us/sample - loss: 1.3865 - val_loss: 1.3648\n",
      "Epoch 27/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3830\n",
      "Epoch 27: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 639us/sample - loss: 1.3830 - val_loss: 1.3602\n",
      "Epoch 28/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3796\n",
      "Epoch 28: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 710us/sample - loss: 1.3796 - val_loss: 1.3616\n",
      "Epoch 29/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3801\n",
      "Epoch 29: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.3801 - val_loss: 1.3623\n",
      "Epoch 30/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3766\n",
      "Epoch 30: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 710us/sample - loss: 1.3766 - val_loss: 1.3642\n",
      "Epoch 31/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3775\n",
      "Epoch 31: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 1.3775 - val_loss: 1.3597\n",
      "Epoch 32/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3836\n",
      "Epoch 32: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 1.3836 - val_loss: 1.3585\n",
      "Epoch 33/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3777\n",
      "Epoch 33: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 709us/sample - loss: 1.3777 - val_loss: 1.3675\n",
      "Epoch 34/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3754\n",
      "Epoch 34: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 632us/sample - loss: 1.3754 - val_loss: 1.3626\n",
      "Epoch 35/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 35: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 682us/sample - loss: 1.3763 - val_loss: 1.3657\n",
      "Epoch 36/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3717\n",
      "Epoch 36: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 640us/sample - loss: 1.3717 - val_loss: 1.3551\n",
      "Epoch 37/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3727\n",
      "Epoch 37: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 707us/sample - loss: 1.3727 - val_loss: 1.3591\n",
      "Epoch 38/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3733\n",
      "Epoch 38: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 698us/sample - loss: 1.3733 - val_loss: 1.3615\n",
      "Epoch 39/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3709\n",
      "Epoch 39: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 630us/sample - loss: 1.3709 - val_loss: 1.3582\n",
      "Epoch 40/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3674\n",
      "Epoch 40: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 21s 670us/sample - loss: 1.3674 - val_loss: 1.3597\n",
      "Epoch 41/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3646\n",
      "Epoch 41: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 634us/sample - loss: 1.3646 - val_loss: 1.3547\n",
      "Epoch 42/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3635\n",
      "Epoch 42: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 1.3635 - val_loss: 1.3583\n",
      "Epoch 43/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3605\n",
      "Epoch 43: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 1.3605 - val_loss: 1.3544\n",
      "Epoch 44/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3628\n",
      "Epoch 44: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 19s 615us/sample - loss: 1.3628 - val_loss: 1.3601\n",
      "Epoch 45/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3616\n",
      "Epoch 45: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 695us/sample - loss: 1.3616 - val_loss: 1.3608\n",
      "Epoch 46/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3618\n",
      "Epoch 46: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 23s 715us/sample - loss: 1.3618 - val_loss: 1.3533\n",
      "Epoch 47/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3587\n",
      "Epoch 47: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 20s 644us/sample - loss: 1.3587 - val_loss: 1.3541\n",
      "Epoch 48/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3592\n",
      "Epoch 48: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 19s 613us/sample - loss: 1.3592 - val_loss: 1.3550\n",
      "Epoch 49/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3570\n",
      "Epoch 49: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 700us/sample - loss: 1.3570 - val_loss: 1.3490\n",
      "Epoch 50/50\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3567\n",
      "Epoch 50: val_loss did not improve from 1.33426\n",
      "31519/31519 [==============================] - 22s 689us/sample - loss: 1.3567 - val_loss: 1.3583\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:08:46.748715: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_41/lstm_cell_185/recurrent_kernel/Assign' id:96176 op device:{requested: '', assigned: ''} def:{{{node lstm_41/lstm_cell_185/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_41/lstm_cell_185/recurrent_kernel, lstm_41/lstm_cell_185/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:08:51.861494: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_25_5/lstm_cell_210/recurrent_kernel/m/Assign' id:103148 op device:{requested: '', assigned: ''} def:{{{node lstm_25_5/lstm_cell_210/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_25_5/lstm_cell_210/recurrent_kernel/m, lstm_25_5/lstm_cell_210/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:08:56.786402: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1968)\n",
      "(1514, 1968)\n",
      "(1644, 1968)\n",
      "(1764, 1968)\n",
      "(1836, 1968)\n",
      "(1699, 1968)\n",
      "(1369, 1968)\n",
      "(1766, 1968)\n",
      "(1619, 1968)\n",
      "(1692, 1968)\n",
      "(1550, 1968)\n",
      "(1920, 1968)\n",
      "(1739, 1968)\n",
      "(1788, 1968)\n",
      "(1812, 1968)\n",
      "(1788, 1968)\n",
      "(970, 1968)\n",
      "(1668, 1968)\n",
      "(1896, 1968)\n",
      "{1: 7.031987783314579, 2: 6.0998418260729785, 4: 8.796502813052564, 5: 5.592161388844238, 6: 4.69383185982961, 8: 8.604664844880244, 9: 6.8619893907731875, 10: 8.280064941327762, 11: 6.970884423726089, 12: 9.735255425172802, 13: 6.6506883137759205, 17: 9.495762739109468, 19: 8.134000964475165, 21: 10.0, 22: 2.3608011573701977, 25: 8.694964796656485, 27: 4.986995659861758, 28: 6.908027648288057, 29: 1.0}\n",
      "Train on 31519 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:12:57.754386: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6156\n",
      "Epoch 1: val_loss improved from inf to 1.39225, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 35s 1ms/sample - loss: 9.6156 - val_loss: 1.3922\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5901\n",
      "Epoch 2: val_loss improved from 1.39225 to 1.36646, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 23s 719us/sample - loss: 9.5901 - val_loss: 1.3665\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.6384\n",
      "Epoch 3: val_loss improved from 1.36646 to 1.36085, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 23s 715us/sample - loss: 9.6384 - val_loss: 1.3608\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.5256\n",
      "Epoch 4: val_loss improved from 1.36085 to 1.35315, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 23s 719us/sample - loss: 9.5256 - val_loss: 1.3531\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4731\n",
      "Epoch 5: val_loss improved from 1.35315 to 1.34873, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 22s 701us/sample - loss: 9.4731 - val_loss: 1.3487\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4627\n",
      "Epoch 6: val_loss improved from 1.34873 to 1.34559, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 20s 622us/sample - loss: 9.4627 - val_loss: 1.3456\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4414\n",
      "Epoch 7: val_loss did not improve from 1.34559\n",
      "31519/31519 [==============================] - 19s 617us/sample - loss: 9.4414 - val_loss: 1.3509\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4181\n",
      "Epoch 8: val_loss did not improve from 1.34559\n",
      "31519/31519 [==============================] - 22s 703us/sample - loss: 9.4181 - val_loss: 1.3461\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4467\n",
      "Epoch 9: val_loss did not improve from 1.34559\n",
      "31519/31519 [==============================] - 22s 714us/sample - loss: 9.4467 - val_loss: 1.3460\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.4125\n",
      "Epoch 10: val_loss did not improve from 1.34559\n",
      "31519/31519 [==============================] - 22s 712us/sample - loss: 9.4125 - val_loss: 1.3525\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3979\n",
      "Epoch 11: val_loss improved from 1.34559 to 1.34028, saving model to ./checkpoints/unknown_person_p26_22.h5\n",
      "31519/31519 [==============================] - 20s 634us/sample - loss: 9.3979 - val_loss: 1.3403\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3884\n",
      "Epoch 12: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 23s 717us/sample - loss: 9.3884 - val_loss: 1.3425\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3818\n",
      "Epoch 13: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 22s 686us/sample - loss: 9.3818 - val_loss: 1.3452\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3955\n",
      "Epoch 14: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 9.3955 - val_loss: 1.3451\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3657\n",
      "Epoch 15: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 20s 632us/sample - loss: 9.3657 - val_loss: 1.3470\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3561\n",
      "Epoch 16: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 20s 638us/sample - loss: 9.3561 - val_loss: 1.3567\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3778\n",
      "Epoch 17: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 22s 706us/sample - loss: 9.3778 - val_loss: 1.3509\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3545\n",
      "Epoch 18: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 21s 674us/sample - loss: 9.3545 - val_loss: 1.3490\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3483\n",
      "Epoch 19: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 22s 692us/sample - loss: 9.3483 - val_loss: 1.3496\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 9.3396\n",
      "Epoch 20: val_loss did not improve from 1.34028\n",
      "31519/31519 [==============================] - 20s 627us/sample - loss: 9.3396 - val_loss: 1.3408\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:20:20.734268: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_25_6/lstm_cell_247/kernel/Assign' id:119556 op device:{requested: '', assigned: ''} def:{{{node lstm_25_6/lstm_cell_247/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_25_6/lstm_cell_247/kernel, lstm_25_6/lstm_cell_247/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:20:26.786038: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_14_6/lstm_cell_236/recurrent_kernel/m/Assign' id:122381 op device:{requested: '', assigned: ''} def:{{{node lstm_14_6/lstm_cell_236/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_14_6/lstm_cell_236/recurrent_kernel/m, lstm_14_6/lstm_cell_236/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31519 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:20:35.843887: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:21:07.668477: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3175"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:21:28.740592: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33458, saving model to ./checkpoints/unknown_person_baseline_2_p26_22.h5\n",
      "31519/31519 [==============================] - 33s 1ms/sample - loss: 1.3175 - val_loss: 1.3346\n",
      "Epoch 2/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3200\n",
      "Epoch 2: val_loss did not improve from 1.33458\n",
      "31519/31519 [==============================] - 20s 628us/sample - loss: 1.3200 - val_loss: 1.3360\n",
      "Epoch 3/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3278\n",
      "Epoch 3: val_loss did not improve from 1.33458\n",
      "31519/31519 [==============================] - 21s 662us/sample - loss: 1.3278 - val_loss: 1.3412\n",
      "Epoch 4/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3265\n",
      "Epoch 4: val_loss did not improve from 1.33458\n",
      "31519/31519 [==============================] - 22s 708us/sample - loss: 1.3265 - val_loss: 1.3400\n",
      "Epoch 5/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3206\n",
      "Epoch 5: val_loss improved from 1.33458 to 1.33421, saving model to ./checkpoints/unknown_person_baseline_2_p26_22.h5\n",
      "31519/31519 [==============================] - 21s 681us/sample - loss: 1.3206 - val_loss: 1.3342\n",
      "Epoch 6/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3177\n",
      "Epoch 6: val_loss improved from 1.33421 to 1.33033, saving model to ./checkpoints/unknown_person_baseline_2_p26_22.h5\n",
      "31519/31519 [==============================] - 22s 713us/sample - loss: 1.3177 - val_loss: 1.3303\n",
      "Epoch 7/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3230\n",
      "Epoch 7: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 21s 665us/sample - loss: 1.3230 - val_loss: 1.3389\n",
      "Epoch 8/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3185\n",
      "Epoch 8: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 22s 703us/sample - loss: 1.3185 - val_loss: 1.3340\n",
      "Epoch 9/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3188\n",
      "Epoch 9: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 20s 630us/sample - loss: 1.3188 - val_loss: 1.3329\n",
      "Epoch 10/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3172\n",
      "Epoch 10: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 19s 609us/sample - loss: 1.3172 - val_loss: 1.3379\n",
      "Epoch 11/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3179\n",
      "Epoch 11: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 19s 606us/sample - loss: 1.3179 - val_loss: 1.3356\n",
      "Epoch 12/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3164\n",
      "Epoch 12: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 21s 651us/sample - loss: 1.3164 - val_loss: 1.3365\n",
      "Epoch 13/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3168\n",
      "Epoch 13: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 22s 691us/sample - loss: 1.3168 - val_loss: 1.3320\n",
      "Epoch 14/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3178\n",
      "Epoch 14: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 20s 623us/sample - loss: 1.3178 - val_loss: 1.3385\n",
      "Epoch 15/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3155\n",
      "Epoch 15: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 20s 621us/sample - loss: 1.3155 - val_loss: 1.3403\n",
      "Epoch 16/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3141\n",
      "Epoch 16: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 19s 607us/sample - loss: 1.3141 - val_loss: 1.3312\n",
      "Epoch 17/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3149\n",
      "Epoch 17: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 21s 671us/sample - loss: 1.3149 - val_loss: 1.3347\n",
      "Epoch 18/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3135\n",
      "Epoch 18: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 23s 741us/sample - loss: 1.3135 - val_loss: 1.3367\n",
      "Epoch 19/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3121\n",
      "Epoch 19: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 20s 650us/sample - loss: 1.3121 - val_loss: 1.3334\n",
      "Epoch 20/20\n",
      "31519/31519 [==============================] - ETA: 0s - loss: 1.3126\n",
      "Epoch 20: val_loss did not improve from 1.33033\n",
      "31519/31519 [==============================] - 21s 675us/sample - loss: 1.3126 - val_loss: 1.3340\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [26]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
